### DISCUSSION TITLE
[Suggestion] Use interactive prompt

### DISCUSSION THREAD

**Discussion by DrMagPie** [ThioJoe/YT-Spammer-Purge#451] [Ideas]
Hi @ThioJoe,

First of all amazing project! Especially love the `report` mode ❤️ 
Getting to the suggestion, the interface method you are using is a bit old fashioned. it does the thing and does it well.
But more modern version would use something like https://github.com/tmbo/questionary for prompting user with choices.

as example that would allow you to replace following bulk of code:
https://github.com/ThioJoe/YT-Spammer-Purge/blob/66b6a893e58466ba6225dae23acef4fffe54b808/YTSpammerPurge.py#L400-L424

with this:
```python
scanMode = questionary.select(
    "Chooses mode you want to run in",
    choices=[
        'chosenvideos',
        'recentvideos',
        'entirechannel',
        'communitypost',
        'commentlist',
        'recentcommunityposts'
    ]
).ask()  # returns value of selection
```

and that should result in more maintainable and readable code.

Best wishes,
Random Person From The Internet 😃

**Comment by TechStudent10** [comment]
I support this. This seems like a really good idea.

**Comment by Firecul** [comment]
Short of a GUI this seems like a good suggestion
